src/components/NewKeyPairForm.tsx

```diff
@@ -0,0 +1,195 @@
+  margin-bottom: 20px;
+`;
+
+const TempErrorEl = styled.div`
+  color: #c00;
+  margin-bottom: 20px;
+`;
+
+interface KeyPairType {
+  publicKey: string;
+  secretKey: string;
+}
+
+interface NewKeyPairFormProps {
+  onClose?: () => void;
+}
+
+export const NewKeyPairForm = ({ onClose }: NewKeyPairFormProps) => {
+  const [acceptedWarning, setAcceptedWarning] = useState(false);
+  const [newKeyPair, setNewKeyPair] = useState<KeyPairType | undefined>();
+  const [keyPairCopyString, setKeyPairCopyString] = useState("");
+  const [isKeyPairCopied, setIsKeyPairCopied] = useState(false);
+  const [confirmSavedSecretKey, setConfirmSavedSecretKey] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const generateNewKeyPair = () => {
+    const keypair = Keypair.random();
+
+    setNewKeyPair({
+      publicKey: keypair.publicKey(),
+      secretKey: keypair.secret(),
+    });
+
+    // Spacing here is important for copied string
+    setKeyPairCopyString(`Public key:
+${keypair.publicKey()}
+Secret key:
+${keypair.secret()}`);
+  };
+
+  const handleContinue = () => {
+    setAcceptedWarning(true);
+    generateNewKeyPair();
+  };
+
+  const handleDone = () => {
+    if (!confirmSavedSecretKey) {
+      setErrorMessage(
+        "Please confirm that you have copied and saved your secret key",
+      );
+      return;
+    }
+
+    handleClose();
+  };
+
+  const handleClose = () => {
+    if (onClose) {
+      onClose();
+    }
+  };
+
+  const handleCopyKeys = () => {
+    setIsKeyPairCopied(true);
+  };
+
+  const toggleConfirmSavedSecretKey = () => {
+    setErrorMessage("");
+    setConfirmSavedSecretKey(!confirmSavedSecretKey);
+  };
+
+  return (
+    <>
+      {/* Show warning */}
+      {!acceptedWarning && (
+        <div>
+          <h2>Generate a new key pair</h2>
+
+          <WarningEl>
+            <h3>
+              ATTENTION: Copying and pasting your secret key is not recommended
+            </h3>
+
+            <ul>
+              <li>
+                By copying and pasting your secret key you are vulnerable to
+                different attacks and scams that can result in your secret key
+                being stolen.
+              </li>
+              <li>Only use this option if you're aware of the risks...</li>
+              <li>
+                Ideally use other authentication methods like a hardware wallet
+                or a browser
+              </li>
+              <li>...</li>
+            </ul>
+          </WarningEl>
+
+          <TempButtonEl onClick={handleContinue}>Continue</TempButtonEl>
+          <TempButtonEl onClick={handleClose}>Cancel</TempButtonEl>
+        </div>
+      )}
+
+      {/* Show generate new key pair form */}
+      {acceptedWarning && (
+        <div>
+          <h2>New key pair</h2>
+
+          <WarningEl>
+            <h3>ATTENTION:</h3>
+
+            <ul>
+              <li>
+                It's really important to keep track of your secret key, and to
```

Comment:
You should get used to typing the right typographical character; that way, you don't have to painstakingly go back and fix them later. See also https://github.com/stellar/product-conventions/blob/master/STYLEGUIDE.md#use-the-right-symbols